lib/srv/discovery/fetchers/db/azure_redis.go

```diff
@@ -45,22 +47,26 @@ func (p *azureRedisPlugin) GetServerLocation(server *armredis.ResourceInfo) stri
 	return azure.StringVal(server.Location)
 }
 
-func (p *azureRedisPlugin) NewDatabaseFromServer(server *armredis.ResourceInfo, log logrus.FieldLogger) types.Database {
+func (p *azureRedisPlugin) NewDatabaseFromServer(ctx context.Context, server *armredis.ResourceInfo, logger *slog.Logger) types.Database {
 	if server.Properties.SSLPort == nil { // should never happen, but checking just in case.
-		log.Debugf("Azure Redis server %v is missing SSL port. Skipping.", azure.StringVal(server.Name))
+		logger.DebugContext(ctx, "Skipping Azure Redis server with missing SSL port", "server", azure.StringVal(server.Name))
 		return nil
 	}
 
 	if !p.isAvailable(server) {
-		log.Debugf("The current status of Azure Redis server %q is %q. Skipping.",
-			azure.StringVal(server.Name),
-			azure.StringVal(server.Properties.ProvisioningState))
+		logger.DebugContext(ctx, "Skipping unavailable Azure Redis server",
+			"server", azure.StringVal(server.Name),
+			"provisioning_state", azure.StringVal(server.Properties.ProvisioningState),
```

Comment:
```suggestion
			"status", azure.StringVal(server.Properties.ProvisioningState),
```

(to be consistent with other places in this file and package.